### PR TITLE
Do not let site crash on menu if there is a namespace mismatch

### LINKED ISSUE #532: BlogConfig matching query does not exist
Ubuntu 18-04 in WSL (Windows 10 Pro)
Following manual instructions from [here](https://github.com/nephila/djangocms-blog/blob/develop/docs/installation.rst)

I missed publish web page step.

After stopping and starting server with `python manage.py runserver`
have non-working app.

> Environment:
> 
> 
> Request Method: GET
> Request URL: http://localhost:8000/en/
> 
> Django Version: 2.2.9
> Python Version: 3.6.8
> Installed Applications:
> ['djangocms_admin_style',
>  'django.contrib.auth',
>  'django.contrib.contenttypes',
>  'django.contrib.sessions',
>  'django.contrib.admin',
>  'django.contrib.sites',
>  'django.contrib.sitemaps',
>  'django.contrib.staticfiles',
>  'django.contrib.messages',
>  'cms',
>  'menus',
>  'sekizai',
>  'treebeard',
>  'djangocms_text_ckeditor',
>  'filer',
>  'easy_thumbnails',
>  'djangocms_bootstrap4',
>  'djangocms_bootstrap4.contrib.bootstrap4_alerts',
>  'djangocms_bootstrap4.contrib.bootstrap4_badge',
>  'djangocms_bootstrap4.contrib.bootstrap4_card',
>  'djangocms_bootstrap4.contrib.bootstrap4_carousel',
>  'djangocms_bootstrap4.contrib.bootstrap4_collapse',
>  'djangocms_bootstrap4.contrib.bootstrap4_content',
>  'djangocms_bootstrap4.contrib.bootstrap4_grid',
>  'djangocms_bootstrap4.contrib.bootstrap4_jumbotron',
>  'djangocms_bootstrap4.contrib.bootstrap4_link',
>  'djangocms_bootstrap4.contrib.bootstrap4_listgroup',
>  'djangocms_bootstrap4.contrib.bootstrap4_media',
>  'djangocms_bootstrap4.contrib.bootstrap4_picture',
>  'djangocms_bootstrap4.contrib.bootstrap4_tabs',
>  'djangocms_bootstrap4.contrib.bootstrap4_utilities',
>  'djangocms_file',
>  'djangocms_icon',
>  'djangocms_link',
>  'djangocms_picture',
>  'djangocms_style',
>  'djangocms_snippet',
>  'djangocms_googlemap',
>  'djangocms_video',
>  'aldryn_apphooks_config',
>  'aldryn_search',
>  'parler',
>  'taggit',
>  'taggit_autosuggest',
>  'meta',
>  'sortedm2m',
>  'djangocms_blog',
>  'mysite']
> Installed Middleware:
> ['cms.middleware.utils.ApphookReloadMiddleware',
>  'django.contrib.sessions.middleware.SessionMiddleware',
>  'django.middleware.csrf.CsrfViewMiddleware',
>  'django.contrib.auth.middleware.AuthenticationMiddleware',
>  'django.contrib.messages.middleware.MessageMiddleware',
>  'django.middleware.locale.LocaleMiddleware',
>  'django.middleware.common.CommonMiddleware',
>  'django.middleware.clickjacking.XFrameOptionsMiddleware',
>  'cms.middleware.user.CurrentUserMiddleware',
>  'cms.middleware.page.CurrentPageMiddleware',
>  'cms.middleware.toolbar.ToolbarMiddleware',
>  'cms.middleware.language.LanguageCookieMiddleware']
> 
> 
> Template error:
> In template /home/ivan/trying-django-cms/first-project/mysite/mysite/templates/base.html, error at line 35
>    BlogConfig matching query does not exist.
>    25 :                 float: left;
>    26 :                 width: 20%;
>    27 :             }
>    28 :         </style>
>    29 :         {% render_block "css" %}
>    30 :     </head>
>    31 :     <body>
>    32 :         {% cms_toolbar %}
>    33 :         <div class="container">
>    34 :             <ul class="nav">
>    35 :                  {% show_menu 0 100 100 100 %} 
>    36 :             </ul>
>    37 :             {% block content %}{% endblock content %}
>    38 :         </div>
>    39 :         {% render_block "js" %}
>    40 :     </body>
>    41 : </html>
>    42 : 
> 
> Traceback:
> 
> File "/home/ivan/trying-django-cms/env/lib/python3.6/site-packages/django/core/handlers/exception.py" in inner
>   34.             response = get_response(request)
> 
> File "/home/ivan/trying-django-cms/env/lib/python3.6/site-packages/django/core/handlers/base.py" in _get_response
>   145.                 response = self.process_exception_by_middleware(e, request)
> 
> File "/home/ivan/trying-django-cms/env/lib/python3.6/site-packages/django/core/handlers/base.py" in _get_response
>   143.                 response = response.render()
> 
> File "/home/ivan/trying-django-cms/env/lib/python3.6/site-packages/django/template/response.py" in render
>   106.             self.content = self.rendered_content
> 
> File "/home/ivan/trying-django-cms/env/lib/python3.6/site-packages/django/template/response.py" in rendered_content
>   83.         content = template.render(context, self._request)
> 
> File "/home/ivan/trying-django-cms/env/lib/python3.6/site-packages/django/template/backends/django.py" in render
>   61.             return self.template.render(context)
> 
> File "/home/ivan/trying-django-cms/env/lib/python3.6/site-packages/django/template/base.py" in render
>   171.                     return self._render(context)
> 
> File "/home/ivan/trying-django-cms/env/lib/python3.6/site-packages/django/template/base.py" in _render
>   163.         return self.nodelist.render(context)
> 
> File "/home/ivan/trying-django-cms/env/lib/python3.6/site-packages/django/template/base.py" in render
>   937.                 bit = node.render_annotated(context)
> 
> File "/home/ivan/trying-django-cms/env/lib/python3.6/site-packages/django/template/base.py" in render_annotated
>   904.             return self.render(context)
> 
> File "/home/ivan/trying-django-cms/env/lib/python3.6/site-packages/django/template/loader_tags.py" in render
>   150.             return compiled_parent._render(context)
> 
> File "/home/ivan/trying-django-cms/env/lib/python3.6/site-packages/django/template/base.py" in _render
>   163.         return self.nodelist.render(context)
> 
> File "/home/ivan/trying-django-cms/env/lib/python3.6/site-packages/django/template/base.py" in render
>   937.                 bit = node.render_annotated(context)
> 
> File "/home/ivan/trying-django-cms/env/lib/python3.6/site-packages/django/template/base.py" in render_annotated
>   904.             return self.render(context)
> 
> File "/home/ivan/trying-django-cms/env/lib/python3.6/site-packages/classytags/core.py" in render
>   153.         return self.render_tag(context, **kwargs)
> 
> File "/home/ivan/trying-django-cms/env/lib/python3.6/site-packages/sekizai/templatetags/sekizai_tags.py" in render_tag
>   89.         rendered_contents = nodelist.render(context)
> 
> File "/home/ivan/trying-django-cms/env/lib/python3.6/site-packages/django/template/base.py" in render
>   937.                 bit = node.render_annotated(context)
> 
> File "/home/ivan/trying-django-cms/env/lib/python3.6/site-packages/django/template/base.py" in render_annotated
>   904.             return self.render(context)
> 
> File "/home/ivan/trying-django-cms/env/lib/python3.6/site-packages/classytags/core.py" in render
>   153.         return self.render_tag(context, **kwargs)
> 
> File "/home/ivan/trying-django-cms/env/lib/python3.6/site-packages/cms/templatetags/cms_tags.py" in render_tag
>   447.             return toolbar.render_with_structure(context, nodelist)
> 
> File "/home/ivan/trying-django-cms/env/lib/python3.6/site-packages/cms/toolbar/toolbar.py" in render_with_structure
>   477.         rendered_contents = nodelist.render(context)
> 
> File "/home/ivan/trying-django-cms/env/lib/python3.6/site-packages/django/template/base.py" in render
>   937.                 bit = node.render_annotated(context)
> 
> File "/home/ivan/trying-django-cms/env/lib/python3.6/site-packages/django/template/base.py" in render_annotated
>   904.             return self.render(context)
> 
> File "/home/ivan/trying-django-cms/env/lib/python3.6/site-packages/classytags/core.py" in render
>   153.         return self.render_tag(context, **kwargs)
> 
> File "/home/ivan/trying-django-cms/env/lib/python3.6/site-packages/classytags/helpers.py" in render_tag
>   88.                 flatten_context(self.get_context(context, **kwargs))
> 
> File "/home/ivan/trying-django-cms/env/lib/python3.6/site-packages/menus/templatetags/menu_tags.py" in get_context
>   144.             nodes = menu_renderer.get_nodes(namespace, root_id)
> 
> File "/home/ivan/trying-django-cms/env/lib/python3.6/site-packages/django/utils/functional.py" in __wrapper__
>   152.                 return getattr(res, method_name)(*args, **kw)
> 
> File "/home/ivan/trying-django-cms/env/lib/python3.6/site-packages/menus/menu_pool.py" in get_nodes
>   226.         nodes = self._build_nodes()
> 
> File "/home/ivan/trying-django-cms/env/lib/python3.6/site-packages/menus/menu_pool.py" in _build_nodes
>   177.                 nodes = menu.get_nodes(self.request)
> 
> File "/home/ivan/trying-django-cms/env/lib/python3.6/site-packages/djangocms_blog/cms_menus.py" in get_nodes
>   51.                     namespace=self.instance.application_namespace
> 
> File "/home/ivan/trying-django-cms/env/lib/python3.6/site-packages/django/db/models/manager.py" in manager_method
>   82.                 return getattr(self.get_queryset(), name)(*args, **kwargs)
> 
> File "/home/ivan/trying-django-cms/env/lib/python3.6/site-packages/django/db/models/query.py" in get
>   408.                 self.model._meta.object_name
> 
> Exception Type: DoesNotExist at /en/
> Exception Value: BlogConfig matching query does not exist.

### PR DESCRIPTION
**Pull request by g3rb3n**
# Description
* When there is a namespace mismatch do not let menu templatetag throw an error
* 532 

## References
Fix #532 

# Checklist

* [x] I have read the [contribution guide](https://django-app-helper.readthedocs.io/en/latest/contributing.html)
* [x] Code lint checked via `inv lint`
* [x] ``changes`` file included (see [docs](https://django-app-helper.readthedocs.io/en/latest/contributing.html#pull-request-guidelines))
* [x] Tests added

### CODE REVIEW

File: tests/test_namespace.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        yield
+
+
+class AdminTest(BaseTest):
+    def setUp(self):
+        User.objects.create_superuser("test_admin", "admin@example.com", "Password123")
+
+    def add_blog_config(self, page, namespace, id):
+        self.client = Client()
+        self.client.login(username="test_admin", password="Password123")
```
Comment (yakky) [inline tests/test_namespace.py:25]: you can use `self._admin_user_username`, `self._admin_user_email`  and `self._admin_user_password` to make the login, an admin user is automatically created and you can drop the `setUp` method above

File: tests/test_namespace.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
+from contextlib import contextmanager
+
+from cms.api import create_page
+from django.test import Client
+from django.utils.encoding import force_str
+
+from .base import BaseTest
+from .test_utils.models import CustomUser as User
+
+try:
```
Comment (yakky) [inline tests/test_namespace.py:10]: this is not needed

File: djangocms_blog/cms_menus.py [hunk 3]
```diff
@@ -43,9 +47,13 @@ def get_nodes(self, request):
...
         if self.instance:
             if not self._config.get(self.instance.application_namespace, False):
-                self._config[self.instance.application_namespace] = BlogConfig.objects.get(
-                    namespace=self.instance.application_namespace
-                )
+                try:
+                    self._config[self.instance.application_namespace] = BlogConfig.objects.get(
+                        namespace=self.instance.application_namespace
+                    )
+                except Exception as e:
```
Comment (yakky) [inline djangocms_blog/cms_menus.py:54]: IDK if it's work to have a stricter check here (`BlogConfig.DoesNotExist` and `AttributeError` should suffice) or keep it broad :thinking:

File: tests/test_namespace.py (hunk 1 above)
Comment (g3rb3n) [reply tests/test_namespace.py:25]: Thank you for the suggestion! Makes live easier.

File: djangocms_blog/cms_menus.py (hunk 3 above)
Comment (g3rb3n) [reply djangocms_blog/cms_menus.py:54]: Totally agree on this one.

File: tests/test_namespace.py (hunk 2 above)
Comment (g3rb3n) [reply tests/test_namespace.py:9]: Agreed

Comment (yakky) [reply tests/test_namespace.py:9]: apparently the push with this change did not make it (I had myself issues with github in the past hour or so)

Comment (g3rb3n) [reply tests/test_namespace.py:9]: Commandline says it is up to date.

I notice that github is changing its ip every few minutes...

Comment (yakky) [reply tests/test_namespace.py:9]: github cowardly refusing to get your commit, apparently :roll_eyes: 

nevermind, I will merge as is and we will cleanup later

**Pull request merged**